### DISCUSSION TITLE
Updated toast design discussion

### DISCUSSION THREAD

**Discussion by Stoppedpuma** [ppy/osu#27305] [Ideas]
Continuation of #25094 as requested by @Joehuu. The PR has been moved to a discussion for proper feedback and discussion. You can see my design proposal [on figma](https://www.figma.com/file/FWgz2MZQEh80lHSlaywtnC/toast?type=design&node-id=0%3A1&mode=design&t=9Q8DrNwnxitp4Ibc-1) or in the image below:

![1708565505](https://github.com/ppy/osu/assets/58333920/7edb6582-3ce7-4e50-8bac-3ba6d151e70d)

**Comment by bdach** [comment]
The spacings are way too tight.

**Reply by Stoppedpuma** [reply]
Does this look better to you?

![1708634431](https://github.com/ppy/osu/assets/58333920/77a8629f-152b-4de7-a31e-2017b55a2e01)

**Reply by bdach** [reply]
I don't think so no. Vertically everything is so squished together compared to horizontally where there's acres of space.

**Reply by Stoppedpuma** [reply]
How about any of these? (98H was the original height for scale of comparing the new ones)

![1708642562](https://github.com/ppy/osu/assets/58333920/8760969e-f3dd-4581-8eb6-2e2943d58490)
![1708643229](https://github.com/ppy/osu/assets/58333920/894b2167-ad80-4b5e-8fbd-f3bb3c7cb887)

**Reply by peppy** [reply]
All still look bad. 125 is probably getting okay, but the icon is padded wrong.

**Reply by Stoppedpuma** [reply]
@peppy I've updated the icon as well as made two more height options, I'll make some different designs as well in case you don't like this design.

![1709315218](https://github.com/ppy/osu/assets/58333920/70fda08c-40b5-4011-84b2-df9c5f5302a7)

**Reply by Stoppedpuma** [reply]
I've experimented with adding the gradient border around the toast that's in other areas with the same shape, This one is kind of mixed on my end but it could be a cool addition. Below is how 140H would look with borders, the bright green could be a nice extra to show when an option is enabled.

![1709602401](https://github.com/ppy/osu/assets/58333920/50b9857a-e7e9-4fe1-a2fc-d635286007cf)

This second image shows 140H with updated padding on the text and bars since I forgot to update it with the new heights. The image above does not contain the new padding but I will update the gradient version if it's the chosen design option.

![1709654393](https://github.com/ppy/osu/assets/58333920/fed049e2-34ec-489f-a499-94b814575aeb)

**Comment by Gabixel** [comment]
I want to give some support and show my iteration (a very basic, random and misaligned sketch) of this design.
Bdach is right, as there's a lot of information scattered (and squished together) vertically, and a lot of space horizontally.
I'd like to give a cue on how I imagine the information a little more spread out, making the content "breathe," and spanning it out on various layers, taking inspiration from the mod select overlay:

![image](https://github.com/ppy/osu/assets/43073074/d30c7f94-0857-4afd-a5b6-b1ef0b7f44a1)

---

I don't know about the use of icons. They certainly try to give uniqueness to each type of notification, but at the same time they take up a lot of space and give little information (and I think peppy follows this thought, from [the way it went with the settings overlay](https://github.com/ppy/osu/pull/26147)).

Maybe you can still insert them but in a more subtle way:

![image](https://github.com/ppy/osu/assets/43073074/ccf59db5-0924-4d48-a303-5820fa893668)

**Reply by Stoppedpuma** [reply]
While these are certainly far from a bad direction I feel it just doesn't feel right for some reason? Like I try to picture these on the upcoming song select and it feels like an element which isn't quite inline with the rest of the design? That's not to say it's bad, it's definitely a good suggestion and might be good to take ideas from going forward. I'll reply with something that I feel fits more inline with the rest of the design to me while using yours as a base idea and see how it looks.

**Reply by Stoppedpuma** [reply]
I've made a few variants which closely follow your design and which I think follow the overall clients design more closely, does anything that I've changed look odd or out of place to you? I mostly used the [upcoming song select design](https://www.figma.com/file/EjS27IBaZhc2qX97lFBUxN/Song-Select-5?type=design&node-id=0%3A1&mode=design&t=btJg4P9cog4wxGbF-1) as reference.

![1708652174](https://github.com/ppy/osu/assets/58333920/fc528531-0205-4ba1-9b04-09c6a5f3c1c7)

**Reply by Gabixel** [reply]
I think it can be a nice start, yes. I had also kind of forgotten about the use of that little line under certain titles. Can be an interesting touch.

I'm just a little afraid, in general, that by making it too much like the interacting interface (e.g. the carousel), it might make the user think that they can click on toast notifications (unless that's actually the future intent, being able to change options by clicking on them).
The same applies to the design of the toggle "pills," which is identical to that in the settings.

**Reply by Stoppedpuma** [reply]
> it might make the user think that they can click on toast notifications (unless that's actually the future intent, being able to change options by clicking on them).

I don't necessarily see an issue with this? It could be helpful in things like large multiple option choices like frame limiter or one that just came off the top of my head while typing this is moving the beatmap offset (=/-) pop up to the overlay and allowing the offset bar to be dragged? It would probably only be useful during breaks but it's an idea at least. @bdach would it be out of the scope to allow the toast to be user interactable?

> design of the toggle "pills," 

The ones in the current toast already looks pretty similar just a little sharper? Those could be an option to use if we want to sacrifice a bit of design fluency assuming everyone else doesn't really care.

**Reply by peppy** [reply]
The paddings are still all off in these. just feels jank.

**Comment by Stoppedpuma** [comment]
I've made some new designs which could work as alternatives if the first design is out of the picture. I'm not very sure B will work at all but I included it in case someone gets an idea from it or something? Design A might face the same problem as the first one with being too tight vertically. Design C is based on the current toast and is the most likely to avoid the vertical cramping issue, I've included two heights as I think the tall one looks better but the text just seems a little bit too spaced depending on how I look at it and it looks somewhat better on the short version. The black background is because it's transparent to follow the current design which I'll probably end up removing the transparency as I like how it looks against the black.

![1709492209](https://github.com/ppy/osu/assets/58333920/c2d6bff5-52fe-46db-ab7a-f046acf221cb)

**Comment by peppy** [comment]
@arflyte

**Comment by arflyte** [comment]
Design notes:
1. These designs is far too busy for it's need. A simple translucent box would work very well. You may layer few more layers with different opacity to form some frames.

2. As for text sizes, we are working with multiples of 4 (eg: 12, 16, 20, 24, 40) on the design side. Lighter weight for the big ones, heavier for the smaller ones.

The lower designs in the latest is the best direction, but the texture is not necessary.

**Reply by Stoppedpuma** [reply]
I've made a comparison chart [here](https://www.figma.com/file/1eTFbquTbmgI0Pz2EbUMgs/Toast-comparison?type=design&node-id=0%3A1&mode=design&t=oxTigfJA39JNXCh6-1), I have a few questions for you as well:

1. Should I be using the alternative style for torus anywhere?

2. The current option text (Borderless in the examples used) is using `C/C2 - Subcontent` as the colour, is this fine to use or should I switch it to another colour such as white?

3. Is there anything that you feel is missing from the design that you would like to see?

Note regarding the 3rd question: I'll give you editing permissions after you view the file as I don't have your email, please feel free to make changes.

**Reply by bdach** [reply]
> A simple translucent box would work very well

We're doing translucency again? Latest designs have been pretty consistently opaque...

**Reply by arflyte** [reply]
Additional design notes:
1. If opaque, it needs to fly in from somewhere to make it work.
2. If translucent, it can fade in or out. This is far more suitable for simple notification toasters I like.

Of course, the above is not absolute. It's a guideline to keep things consistent.
